### PR TITLE
Chameleon clothing update, this time without warnings

### PR DESCRIPTION
**Pull request by Alliostra**
## About The Pull Request

https://github.com/discordia-space/CEV-Eris/pull/7072
but doesn't cause warnings anymore (the horror!)

now ~~marginally~~ much less convoluted as well, thanks to Kirov.
Also cleans up some chameleon clothing items in general and removed some things you can't disguise chameleon items as.

## Why It's Good For The Game

ask nestor

## Changelog
:cl:
add: added the ability to quickly switch between different loadouts for chameleon clothes if the chameleon jumpsuit is worn.
add: Added the chameleon satchel
fix: chameleon items will no longer block other clothing sprites from appearing when they weren't supposed to
tweak: chameleon clothes are now made of plastic instead of bioatter
tweak: removed some things you can't disguise chameleon items as.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/storage/uplink_kits.dm [hunk 1]
```diff
@@ -129,14 +129,15 @@
 
 /obj/item/storage/box/syndie_kit/chameleon
 	name = "chameleon kit"
-	desc = "Comes with all the clothes you need to impersonate most people.  Acting lessons sold seperately."
+	desc = "Comes with all the clothes you need to impersonate most people. Acting lessons sold seperately. Wearing the uniform will allow fot quick switching between appearances."
```
Comment (hyperioo) [inline code/game/objects/items/weapons/storage/uplink_kits.dm:132]:
```suggestion
	desc = "Comes with all the clothes you need to impersonate most people. Acting lessons sold separately. Wearing the uniform will allow for quick switching between appearances."
```

**Review by hyperioo — APPROVED** [review]

File: code/modules/clothing/chameleon.dm [hunk 2]
```diff
@@ -76,6 +127,195 @@ obj/item/clothing/disguise(newtype, mob/user)
...
 
+//********************************************
+//**Chameleon Jumpsuit loadout functionality**
+//********************************************
+
+/obj/item/clothing/under/chameleon/proc/set_single_loadout_slot(itemtype, loadout)
+	var/obj/item/currentitem = new itemtype(null)
+	var/chameleon_choice
+	var/obj/item/item_to_disguise
+	if(istype(currentitem, /obj/item/clothing/under))
```
Comment (drexample) [inline code/modules/clothing/chameleon.dm:138]: Jesus Christ, just use an associative list

Comment (Alliostra) [reply code/modules/clothing/chameleon.dm]: adressed.

File: code/modules/clothing/chameleon.dm [hunk 3]
```diff
@@ -76,6 +125,198 @@ obj/item/clothing/disguise(newtype, mob/user)
...
+			chameleon_choice = G	
+			chameleon_choice = G.clothing_choices
+			item_to_disguise = input("","Choose a disguise") in chameleon_choice
+			loadout["headset"] = chameleon_choice[item_to_disguise]
+
+
+/obj/item/clothing/under/chameleon/proc/disguise_as_loadout(mob/user, loadout)
+	var/obj/item/A
+	for(var/obj/item/clothing/suit/chameleon/F in user.get_contents())
+		A = F
```
Comment (drexample) [inline code/modules/clothing/chameleon.dm:201]: why?

**Review by drexample — CHANGES_REQUESTED** [review]

**Comment by drexample** [comment]
Jesus this PR can be made five times smaller by giving all chameleon items a var or a component that identifies them as chameleon. This is fucking yandere dev tier code structure.

**Comment by Alliostra** [comment]
> Jesus this PR can be made five times smaller by giving all chameleon items a var or a component that identifies them as chameleon. This is fucking yandere dev tier code structure.

Thank you, this did not occur to me.

File: code/modules/clothing/chameleon.dm (hunk 3 above)
Comment (SirRichardFrancis) [reply code/modules/clothing/chameleon.dm:201]: image

Comment (Alliostra) [reply code/modules/clothing/chameleon.dm]: addressed, I hope

**Review by drexample — APPROVED** [review]

**Pull request merged**